sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -88,7 +88,8 @@ class V2ExpressionBuilder(e: Expression, isPredicate: Boolean = false) {
       } else {
         None
       }
-    case Cast(child, dataType, _, true) =>
+    case Cast(child, dataType, _, ansiEnabled)
+      if Cast.canUpCast(child.dataType, dataType) || ansiEnabled =>
```

Comment:
nit: 4-space indentation?

Reply:
+1 for @huaxingao 's comment. 

In addition, let's switch the predicate order, @beliefer .
```scala
- if Cast.canUpCast(child.dataType, dataType) || ansiEnabled =>
+ if ansiEnabled || Cast.canUpCast(child.dataType, dataType) =>
```

Reply:
Got it.